mongoengine/base/metaclasses.py

```diff
@@ -334,12 +333,16 @@ def __new__(cls, name, bases, attrs):
 
         meta.merge(attrs.get('_meta', {}))  # Top level meta
 
-        # Only simple classes (direct subclasses of Document)
-        # may set allow_inheritance to False
+        # Only simple classes (i.e. direct subclasses of Document) may set
+        # allow_inheritance to False. If the base Document allows inheritance,
+        # none of its subclasses can override allow_inheritance to False.
         simple_class = all([b._meta.get('abstract')
                             for b in flattened_bases if hasattr(b, '_meta')])
-        if (not simple_class and meta['allow_inheritance'] is False and
-                not meta['abstract']):
+        if (
+            not simple_class and
+            meta['allow_inheritance'] is False and
```

Comment:
`allow_inheritance` can be True, False, and None.  True means "allow inheritance", False means "don't allow inheritance", None means "do whatever your parent does, or don't allow inheritance if you're a top-level class". I'll make sure to document this in the right place. The rest of the places can deal with ambiguous False value properly.